core/src/main/java/org/apache/iceberg/SnapshotProducer.java

```diff
@@ -423,21 +423,23 @@ public void commit() {
     try {
       LOG.info("Committed snapshot {} ({})", newSnapshotId.get(), getClass().getSimpleName());
 
-      // at this point, the commit must have succeeded. after a refresh, the snapshot is loaded by
-      // id in case another commit was added between this commit and the refresh.
-      Snapshot saved = ops.refresh().snapshot(newSnapshotId.get());
-      if (saved != null) {
-        cleanUncommitted(Sets.newHashSet(saved.allManifests(ops.io())));
-        // also clean up unused manifest lists created by multiple attempts
-        for (String manifestList : manifestLists) {
-          if (!saved.manifestListLocation().equals(manifestList)) {
-            deleteFile(manifestList);
+      if (!canSkipCleanupAfterCommitSuccess()) {
+        // at this point, the commit must have succeeded. after a refresh, the snapshot is loaded by
+        // id in case another commit was added between this commit and the refresh.
+        Snapshot saved = ops.refresh().snapshot(newSnapshotId.get());
```

Comment:
Can the skipped `ops.refresh()` affect some next operation on the table?

Reply:
If it does, it is undefined behavior. Currently if the refresh fails the operation does not fail it instead logs a warning and continues on. For that reason, seems safe to skip the refresh

Reply:
No it shouldn't. Typically the pattern in Iceberg, is if at some point someone wants the latest metadata there needs to be an explicit refresh. Nothing should rely on something being refreshed at some earlier point.